shared/middleware/files_api.rb

```diff
@@ -855,7 +855,8 @@ def files_put_file(encrypted_channel_id, filename, body)
     if THUMBNAIL_FILENAME == filename
       storage_apps = StorageApps.new(storage_id('user'))
       project_type = storage_apps.project_type_from_channel_id(encrypted_channel_id)
-      if MODERATE_THUMBNAILS_FOR_PROJECT_TYPES.include? project_type
+      content_moderation_enabled = !storage_apps.content_moderation_disabled?(encrypted_channel_id)
+      if MODERATE_THUMBNAILS_FOR_PROJECT_TYPES.include?(project_type) && content_moderation_enabled
```

Comment:
As written, we're performing a DB query to check if moderation is enabled for the project before we check if moderation is disabled for the entire project type.  Let's switch this around so we only do that query if the project type uses automated moderation, which probably means writing some new helpers:

```rb
if moderate_type?(project_type) && moderate_channel?(encrypted_channel_id)
```

Reply:
I like this suggestion and will do in a follow up. Thanks! I want to make sure Josh has functional buttons ASAP if he needs them.